### ISSUE TITLE
Support for NPM7 workspaces (monorepo)

### ISSUE THREAD

**Issue by restfulhead** [intuit/auto#2103]
NPM 7 has introduced [workspaces](https://docs.npmjs.com/cli/v7/using-npm/workspaces) which allows us to manage multiple packages inside one repository without any external dependencies such as Lerna.

It would be fantastic if the auto npm implementation would support this. Basically it would need to parse the `workspaces` array in the root `package.json` and then run the commands for each workspace, e.g. `npm publish -w my-first-module`.

Could this be added?

**Comment by restfulhead** [comment]
I'm currently using the following workaround. Might be interesting for others:

```
  VERSION=`auto version`
  if [ ! -z "$VERSION" ]; then
    auto changelog
    npm version $VERSION -ws
    git add -A
    git commit -m "chore(release): $VERSION [skip ci]"
    npm version $VERSION -m "chore(release): %s [skip ci]"
    git push *** HEAD:main --tags
    VERSION_NO=`node -pe "require('./package.json').version"`
    auto release --use-version $VERSION_NO

    for workspace in $workspaces
    do
      npm publish $workspace/dist
    done

  else 
    echo 'Found skip release label, exiting'
    exit 1
  fi
```

This performs versioning, changelog, GitHub release at the root, but still publishes each workspace package to NPM individually. It works, but I'd much rather have independent releases :-)

**Comment by tomdavidson** [comment]
Looks like there is no abstraction around lerna but the software is small enough the change wouldn't be that big.

What about a workaround that copied npm workspaces to lerna packages in a lerna.json so auto could still detect its a monorepo and get each workspace? Then override the lerna execution in a hook?

**Comment by tomdavidson** [comment]
Actually what/who is using the exported [getLernaPackages](https://github.com/intuit/auto/blob/main/packages/core/src/auto.ts#L2316) ? 

Rather than the npm equivalent of  `lerna ls -pla` we coud read the package.json and the lerna.json file. Reading the files rather than invoking a cli will give support whatever suppose the workspace object in package.json

**Comment by tomdavidson** [comment]
@restfulhead  
what about including a lerna.json file even if you dont use the the cli and configure it with  `"useWorkspaces": true`

**Comment by restfulhead** [comment]
@tomdavidson I assumed that doing so would run Lerna commands. If it doesn't require Lerna at all, then that would be fine, yeah.

**Comment by NiccoloOlivieriAchille** [comment]
Any news on this? More than one year and still no "npm workspaces" integration...

**Comment by hipstersmoothie** [comment]
The mono repo support depends heavily on lerna's versioning and publishing. You should need a very minimal setup like @tomdavidson said. It should work fine with npm workspaces. I don't think the npm cli has similar features

**Comment by hipstersmoothie** [comment]
yeah the npm plugin is intentionally designed to let lerna do a lot of the heavy lifting. really you only need install it and add the configuration

Your use case is def not the intended flow for the npm plugin. I could certainly see that workflow being useful. I dream of someday making the mono repo functionality native to all different types of auto plugins. But really don't have the time
